internals/overlord/state/state.go

```diff
@@ -333,11 +333,22 @@ func (s *State) Cache(key, value interface{}) {
 
 // NewChange adds a new change to the state.
 func (s *State) NewChange(kind, summary string) *Change {
+	return s.NewChangeWithNoticeData(kind, summary, nil)
+}
+
+// NewChangeWithNoticeData adds a new change to the state, adding in any provided extra data.
```

Comment:
Let's be a bit more specific: the data is specifically data for the change-update notices, not data for the change itself. I'd say we should call this `noticeData` and tweak the doc comment.